ext/bin/check-command-perf

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+usage() {
+    cat <<EOF
+Usage: [PDB_JAR=JAR] PDBBOX=BOX_DIR $(basename "$0") STOCKPILE_ARCHIVE"
+
+  Unarchive STOCKPILE_ARCHIVE into $PDBBOX/var/stockpile/cmd/q and run
+  pdb to time how long it takes to consume the stockpile files. The result
+  of this test can be seen in the console.
+
+  The version of pdb will be the one run via ./pdb which defaults to
+  the uberjar in target/.
+
+EOF
+}
+
+misuse() { usage 1>&2; exit 2; }
+
+jar="${PDB_JAR:-target/puppetdb.jar}"
+
+if [ -z "$PDBBOX" ]; then
+    echo "PDBBOX is not set in the environment. Do you have a pdbbox created?" 1>&2
+    misuse
+fi
+
+if [ "$#" -le 0 ]; then
+    echo "A STOCKPILE_ARCHIVE wasn't provided." 1>&2
+    misuse
+fi
+
+if [ ! -e "$jar" ]; then
+    printf 'Unable to find the puppetdb jar %q; have you run "lein uberjar"?\n' \
+      "$jar" 1>&2
+    exit 2
+fi
+
+tmpdir=''
+pdb_id=''
+
+on-exit() {
+    if [ "$tmpdir" ]; then rm -rf "$tmpdir"; fi
+    if [ "$pdb_id" ]; then kill "$pdb_id" &> /dev/null; fi
+}
+trap on-exit EXIT
+
+cmds-processed() {
+    local mbeans='http://localhost:8080/metrics/v1/mbeans'
+    curl -sSX GET "$mbeans/puppetlabs.puppetdb.mq:name=global.$1" | jq $2
+}
+
+target_dir="$PDBBOX/var/stockpile/cmd/q"
+stockpile=""
+
+if [ "$(ls "$target_dir" | wc -l)" -gt 0 ]; then
+    echo "$target_dir is not empty. Please make sure the dir is empty before proceeding." 1>&2
+    exit 2
+fi
+
+if [ -z "$1" ]; then
+    echo "An archive that contains the stockpile must be provided!" 1>&2
+    misuse
+else
+    stockpile="$1"
+fi
+
+printf 'Unpacking %q into %q\n' "$stockpile" "$target_dir" 1>&2
+
+tar zxf "$stockpile" -C "$target_dir"
+commands_count="$(ls "$target_dir" | wc -l)"
+
+echo "Running PDB" 1>&2
+
+tmpdir="$(mktemp -d "tmp-dir-test-command-performance")"
+tmpdir="$(cd "$tmpdir" && pwd)"
+
+./pdb services -c "$PDBBOX/conf.d" >"$tmpdir/log.txt" 2>&1 &
+pdb_id=$!
+
+echo "Waiting for PDB to start..." 1>&2
+
+while ! grep -Fq "Finished database garbage collection" "$tmpdir/log.txt"; do
+    sleep 1
+    if [ "$(grep -F "Error during service start!!!" "$tmpdir/log.txt")" ]; then
```

Comment:
Unfortunately if the database server is not started, this will be stuck in a loop.  You also need to check for `Will retry database connection after temporary failure: java.sql.SQLTransientConnectionException: PDBMigrationsPool: default - Connection is not available, request timed out after 3006ms.`